server.go

```diff
@@ -2048,17 +2048,36 @@ func (s *server) peerTerminationWatcher(p *peer) {
 // nextPeerBackoff computes the next backoff duration for a peer's pubkey using
 // exponential backoff. If no previous backoff was known, the default is
 // returned.
-func (s *server) nextPeerBackoff(pubStr string) time.Duration {
+func (s *server) nextPeerBackoff(pubStr string,
+	startTime time.Time) time.Duration {
+
 	// Now, determine the appropriate backoff to use for the retry.
 	backoff, ok := s.persistentPeersBackoff[pubStr]
 	if !ok {
 		// If an existing backoff was unknown, use the default.
 		return defaultBackoff
 	}
 
-	// Otherwise, use a previous backoff to compute the
-	// subsequent randomized exponential backoff duration.
-	return computeNextBackoff(backoff)
+	// If the peer failed to start properly, we'll just the previous backoff
+	// to compute the subsequent randomized exponential backoff duration.
+	// This will roughly double on average.
+	if startTime == lnpeer.StartTimeFailed {
+		return computeNextBackoff(backoff)
+	}
+
+	// The peer succeeded in starting. We'll reduce the timeout duration
+	// by the length of the connection before applying randomized
+	// exponential backoff. We'll only apply this if:
+	//   backoff - connDuration > defaultBackoff
+	connDuration := time.Now().Sub(startTime)
+	if backoff+defaultBackoff > connDuration {
```

Comment:
Shouldn't this be `backoff - connDuration > defaultBackoff` as stated in the comment?

Could also simplify things a bit further:
```
backoff -= connDuration
if backoff < defaultBackoff {
    return defaultBackoff
}

return computeNextBackoff(backoff)
```